lib/bolt/plugin/module.rb

```diff
@@ -158,20 +158,20 @@ def process_params(task, opts)
         # opts are passed directly from inventory but all of the _ options are
         # handled previously. That may not always be the case so filter them
         # out now.
-        _meta, params = opts.partition { |key, _val| key.start_with?('_') }.map(&:to_h)
-        metaparams = {}
+        meta, params = opts.partition { |key, _val| key.start_with?('_') }.map(&:to_h)
+        validate_params(task, params)
```

Comment:
This validation needs to happen after merging with config, so that keys configured in `config` instead of the inventory are included in validation

Reply:
Also needs a rubocop run

Reply:
shouldn't those keys have been validated when the config was loaded? We definitely cannot validate after `_config` is added.

Reply:
We merge config from bolt.yaml with config from inventory if no config is defined in `bolt_plugin.json`, so we need to validate the merged config+inventory in that case